Widen the test for configure regeneration in Travis

dra27 opened the pull request:
`configure` depends on `VERSION` and should be generated whenever the compiler version is bumped.

@damiendoligez - I think it's OK that this configure could be updated in the "last commit"/"first commit" commits, therefore?

With this PR, the Travis branch build of trunk would have (correctly) failed after 4.09 was branched.

I've also fixed an innocuous error message displayed in the `check-typo` job of Travis branch builds - GitHub makes the diff look much more drastic than it is (comment inline)

cc @avsm

dra27 committed:
Check all necessary paths for configure

dra27 committed:
Fix error message displayed on Travis push

dra27 commented on tools/ci/travis/travis-ci.sh:
```diff
@@ -37,13 +37,20 @@ else
+    exit 1
+  fi
+else
+  if [[ $TRAVIS_COMMIT != $(git rev-parse FETCH_HEAD) ]] ; then
```
From here on is identical to the deleted chunk above, just indented.

dra27 commented on tools/ci/travis/travis-ci.sh:
```diff
@@ -37,13 +37,20 @@ else
-  else
-    echo "TRAVIS_COMMIT does not exist; setting to FETCH_HEAD"
-    TRAVIS_COMMIT=$FETCH_HEAD
+if [[ $TRAVIS_EVENT_TYPE = 'push' ]] ; then
```
There's no FETCH_HEAD for a push, so this just ensure that `TRAVIS_COMMIT` does exist in the repo. Prior to this change, there was an error message from git complaining about the lack of the `FETCH_HEAD` followed by the "TRAVIS_COMMIT exists, so going with it" message.

dra27 commented:
(@avsm - this doesn't resolve the fact that `autogen` doesn't work on macOS, which I'll deal with separately)

damiendoligez commented:
> @damiendoligez - I think it's OK that this configure could be updated in the "last commit"/"first commit" commits, therefore?

Indeed. Could you add the corresponding instructions in section 5 of `tools/release-checklist`?

dra27 committed:
Update release-checklist

damiendoligez reviewed (approved):
Looks good but `autogen` doesn't work on the Release Manager's primary machine... I believe you have some work in progress on that, or should I patch it myself?

dra27 commented:
I have the diagnosis, but not yet a fix for macOS! If you've already got a fix, please go for it! (otherwise I will look at it tomorrow)

avsm commented:
The very very quick fix for `autogen` is to replace `sed` with `gsed` on macOS. Amazingly, there seems to be no consensus on macOS about how to determine/get GNU sed, so I'm not sure how best to upstream this...

dra27 commented:
Hang on, is the problem here just the `-i`? In which case, the fix is easy and I’ll push it later, as writing `sh` on my phone is alas not easy...

avsm commented:
No, the extended regexp syntax also fails.  Could you paste an example of the things you are trying to filter out? `grep` might be portable enough for this.

dra27 committed:
Force autoconf to run

dra27 committed:
Eliminate use of -i in autogen

-i without writing a backup is not portable.

dra27 committed:
Remove GNUism from autogen

Relative addressing is a GNU sed extension.

dra27 commented:
OK, that should fix it - `-i` without a backup is not portable, so I've used a temporary file instead. Relative addressing (the `+8`) is an obvious feature, and therefore a GNUism, not Posix, so I've switched to the much more readable `N;N;N;N;N;N;N;N;d;`!

I also added the `-f` flag to `autoconf` to ensure it actually recompiles `configure` even when it thinks its unecessary (#8639 restores the dependency-based rebuilding via `make configure`)

I think this is good to go, therefore?

dra27 commented:
Rebased to account for the change in #8726 - @damiendoligez, are you happy with the changes to `autogen` since your original approval?

damiendoligez commented:
I'm happy with the autogen changes. I'll let you merge or squash as you prefer.

dra27 commented:
Thanks!
